### PR TITLE
Integrate cargo clippy check

### LINKED ISSUE #239: Integrate cargo clippy check into CI
Along with `cargo fmt`, we shall integrate cargo [clippy](https://github.com/rust-lang/rust-clippy) check into CI (GA) to achieve better static code analysis.

Medium Prio.

### PR DESCRIPTION
**Pull request by Kailai-Wang**
resolves #239 

This PR:
- integrate `cargo clippy` check into GA
- fix the discovered clippy warnings
- cargo update

### CODE REVIEW

File: node/src/chain_spec.rs [hunk 1]
```diff
@@ -265,7 +265,7 @@ fn generate_genesis(
 				.map(|(acc, aura)| {
 					(
 						acc.clone(),                                      // account id
-						acc.clone(),                                      // validator id
+						acc,                                              // validator id
```
Comment (Satoshi-Kusumoto) [inline node/src/chain_spec.rs:268]: It's clippy who discovered all these clone issues?

File: node/Cargo.toml [hunk 2]
```diff
@@ -61,14 +61,14 @@ frame-rpc-system = { package = "substrate-frame-rpc-system", git = "https://gith
...
 # Cumulus dependencies
-cumulus-client-cli = { git = "https://github.com/paritytech/cumulus.git", branch = "polkadot-v0.9.12" }
-cumulus-client-consensus-aura = { git = "https://github.com/paritytech/cumulus.git", branch = "polkadot-v0.9.12" }
-cumulus-client-consensus-relay-chain = { git = "https://github.com/paritytech/cumulus.git", branch = "polkadot-v0.9.12" }
-cumulus-client-consensus-common = { git = "https://github.com/paritytech/cumulus.git", branch = "polkadot-v0.9.12" }
-cumulus-client-collator = { git = "https://github.com/paritytech/cumulus.git", branch = "polkadot-v0.9.12" }
-cumulus-client-service = { git = "https://github.com/paritytech/cumulus.git", branch = "polkadot-v0.9.12" }
-cumulus-client-network = { git = "https://github.com/paritytech/cumulus.git", branch = "polkadot-v0.9.12" }
-cumulus-primitives-core = { git = "https://github.com/paritytech/cumulus.git", branch = "polkadot-v0.9.12" }
+cumulus-client-cli = { git = "https://github.com/paritytech/cumulus", branch = "polkadot-v0.9.12" }
```
Comment (Satoshi-Kusumoto) [inline node/Cargo.toml:64]: Any reasons?

Comment (Satoshi-Kusumoto) [reply node/Cargo.toml:64]: Unify the code?

File: Cargo.toml [hunk 3]
```diff
@@ -4,5 +4,6 @@ panic = 'unwind'
 [workspace]
 members = [
     'node',
+    'primitives',
```
Comment (Satoshi-Kusumoto) [inline Cargo.toml:7]: This is in order that clippy can reach there?

**Review by Satoshi-Kusumoto — APPROVED** [review]

File: node/src/chain_spec.rs (hunk 1 above)
Comment (Kailai-Wang) [reply node/src/chain_spec.rs:268]: > It's clippy who discovered all these clone issues?

yep

File: node/Cargo.toml (hunk 2 above)
Comment (Kailai-Wang) [reply node/Cargo.toml:64]: > Unify the code?

yep, consistent with other git path
Besides that, vscode always try to normalize the `Cargo.lock` making this file appear in the changed files again and again

File: Cargo.toml (hunk 3 above)
Comment (Kailai-Wang) [reply Cargo.toml:7]: > This is in order that clippy can reach there?

yes and fmt too, they rely on workspace members

**Pull request merged**